dashboard/app/views/levels/editors/_gamelab.html.haml

```diff
@@ -80,6 +80,35 @@
   =f.label 'Disable code debugger (Console is still enabled)'
   = boolean_check_box f, :debugger_disabled
 
+.field
+  = f.label 'Immediately show the results of setup code changes in the playspace'
+  :ruby
+    selector = f.select :auto_run_setup, options_for_select([
+      ['none', nil],
+      ['Run the draw loop once', SharedConstants::GAMELAB_AUTORUN_OPTIONS.draw_loop],
+      ['Call drawSprites()', SharedConstants::GAMELAB_AUTORUN_OPTIONS.draw_sprites],
+      ['Run some other JavaScript', SharedConstants::GAMELAB_AUTORUN_OPTIONS.custom],
+    ], selected: @level.auto_run_setup)
+  = selector
+  = f.label 'Custom setup code'
+  = f.text_field :custom_setup_code
+  :javascript
+    (function () {
+      var autoRunSetup = document.getElementById('level_auto_run_setup');
+      var customSetupCode = document.getElementById('level_custom_setup_code');
+      var changeHandler = function () {
+        if (autoRunSetup.value === 'CUSTOM') {
+          customSetupCode.previousElementSibling.style.display = '';
+          customSetupCode.style.display = '';
+        } else {
+          customSetupCode.previousElementSibling.style.display = 'none';
+          customSetupCode.style.display = 'none';
+        }
+      }
+      autoRunSetup.onchange = changeHandler;
+      changeHandler();
+    })();
```

Comment:
JavaScript in HAML?  When [levelbuilder_gamelab.js](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/sites/studio/pages/levelbuilder_gamelab.js) already exists?

![boo](https://user-images.githubusercontent.com/1615761/38954270-11a67d56-4306-11e8-9282-e37c0f17d00c.gif)

Reply:
I am ashamed